src/main/cpp/blaze_util.cc

```diff
@@ -74,6 +76,26 @@ bool GetNullaryOption(const char *arg, const char *key) {
   return true;
 }
 
+std::vector<std::string> SearchNaryOption(const vector<string>& args,
+                              const char *key) {
+  vector<std::string> values;
+  for (vector<string>::size_type i = 0; i < args.size(); ++i) {
+    if (args[i] == "--") {
```

Comment:
I'm not sure this is correct. Have you tried running commands like `bazel --host_jvm_args -- --host_jvm_args=-Xmx5g build --bes_keywords -- --keep_going :BUILD`? That works for me, so I think that `--` does *not* unconditionally end flag parsing.

Reply:
Thanks! will check. I'd guess it should still work because it has worked in the past for a single `--bazelrc`.

Reply:
Hi @janakdr, do you mean
```
bazel --host_jvm_args -- --host_jvm_args=-Xmx5g build --bes_keywords -- --keep_going :BUILD
```
or 
```
bazel --host_jvm_args=-Xmx5g build --bes_keywords -- --keep_going :BUILD
```
but in either case, the behavior is consistent before and after the change.

Reply:
With this change:
```
$ bazel-bin/src/bazel --host_jvm_args -- --host_jvm_args=-Xmx5g build --bes_keywords -- --keep_going :BUILD
WARNING: Running Bazel server needs to be killed, because the startup options are different.
Starting local Bazel server and connecting to it...
Server crashed during startup. Now printing /private/var/tmp/_bazel_yic/5a517de0ca1edbc0f94cf017f53ee5a1/server/jvm.out
Unrecognized option: --
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
$ echo $?
37
```
Without this change:
```
$ bazel --host_jvm_args -- --host_jvm_args=-Xmx5g build --bes_keywords -- --keep_going :BUILD
Starting local Bazel server and connecting to it...
Server crashed during startup. Now printing /private/var/tmp/_bazel_yic/5a517de0ca1edbc0f94cf017f53ee5a1/server/jvm.out
Unrecognized option: --
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
$ echo $?
37
```

Reply:
My point is that `--` does not necessarily indicate the end of options parsing, whereas your code short-circuits if it ever encounters a `--`. It's not that your code changes current behavior. But it doesn't do what the user would expect in this case, although admittedly it's not a very common one.

Reply:
Thanks for clarifying, @janakdr. Yeah since I wasn't aware of all the possible scenarios here, I decided to mimic the existing logic in `SearchUnaryOption` in order to minimize the risk, so essentially there is no change how to treat `--`.

For the case described here, do you think it would be better to be addressed in an separate issue? I'd also like to understand what the expected outcome would be.

Reply:
Yes, it looks like the C++ code is fundamentally different in how it parses than the Java code is, which is unfortunate. You're correct that this isn't a new bug in this code. What I would expect is the Java behavior, which is able to distinguish between `--` as the argument to a flag or as the signal to end options parsing.

Reply:
Good point. Corrected.